packages/ui-react-utils/src/callRenderProp.ts

```diff
@@ -22,37 +22,53 @@
  * SOFTWARE.
  */
 
-import React from 'react'
+import React, {
+  ClassicComponent,
+  ClassicComponentClass,
+  ClassType,
+  ComponentClass,
+  ComponentState,
+  FunctionComponent,
+  ReactHTML,
+  ReactNode,
+  ReactSVG
+} from 'react'
 
 /**
  * ---
  * category: utilities/react
  * ---
  * Evaluate a provided value to create a renderable React element.
  * @module callRenderProp
- * @param {ReactElement|ReactClass|function} value
- * @param {object} props
- * @return {ReactElement} A renderable React element
+ * @param value
+ * @param props
  */
-function callRenderProp(
-  value: any, // TODO type this better
-  props = {}
+function callRenderProp<P>(
+  value:
+    | keyof ReactHTML
+    | keyof ReactSVG
+    | FunctionComponent<P>
+    | ClassType<
+        P,
+        ClassicComponent<P, ComponentState>,
+        ClassicComponentClass<P>
+      >
+    | ComponentClass<P>
+    | ReactNode,
+  props: P = {} as P
 ) {
   if (typeof value === 'function') {
     // note: this part is needed for React 16
     // TODO: Simplify this when we drop React 15
     // In react 16, `createElement` accepts a function. In react 15 we get an
     // error on rendering the result. Evaluate the function here if it is not a
     // react component.
-    if (
-      !(
-        value.prototype && // fat arrow functions don't have a prototype
-        value.prototype.isReactComponent
-      )
-    ) {
-      return value(props)
+    // fat arrow functions don't have a prototype
+    if (!(value.prototype && value.prototype.isReactComponent)) {
+      return (value as any)(props)
     }
-    return React.createElement(value, props)
+    // TODO type 'value' properly
+    return React.createElement<P>(value as any, props)
```

Comment:
The `value` prop here is a nightmare to type. `React.createElement` does not accept `ReactNode`, yet we seem to be passing `ReactNode`s to this method regularly.